### PR TITLE
Sean & Scott Rebased

### PR DESCRIPTION
**Pull request by beansrowning**
## What

Incorporating changes across three branches back into master to get things straight again.

@scotthlee It looks like your branch was pretty stale when you started working on things, so I made some effort to reflow your commits into master and included my own and some HP tuning stuff.

### Basically

master -> sean
hp_tuning -> sean
sean -> scott (interactive rebase)

## TODO  

I've not tested anything thoroughly and hadn't walked through your branch to see how everything worked, so you might take a gander.

It looks like you did some concurrent development in analysis.py that existed on master more-or-less, but I worked in all the additional functionality (specifying cores, model, etc).

### CODE REVIEW

File: python/feature_tokenization.py [hunk 1]
```diff
@@ -46,17 +47,17 @@
 n_patients = trimmed_seq["medrec_key"].nunique()
 
 # Ensure we're sorted
-trimmed_seq.sort_values(["medrec_key", "pat_key", "dfi"], inplace=True)
```
Comment (beansrowning) [inline python/feature_tokenization.py:49]: @scotthlee This was kind of an oopsie on my part. I assumed somewhat naively that this was the correct sorting, but actually, it shuffled the days from index. I expected it to have a bigger impact on the LSTM than it actually did, though. We should still probably re-run

File: python/feature_tokenization.py [hunk 2]
```diff
@@ -8,9 +8,9 @@
 
 from sklearn.feature_extraction.text import CountVectorizer
 
-# Setting top-level parameters
+# %% Setting top-level parameters
 MIN_DF = 5
-NO_VITALS = True
```
Comment (beansrowning) [inline python/feature_tokenization.py:13]: @scotthlee I also experimented with just dumping everything in (including vitals). We can revert before final depending on what we'd like

File: python/feature_tokenization.py [hunk 3]
```diff
@@ -161,7 +178,15 @@
 pat_inpat = [[inpat_dict[id] for id in np.unique(df.values)]
              for _, df in grouped_pat_keys]
 
-# Adding age at each visit
+# %% Adding the ICU indicator
+icu_pats = misa_data[misa_data.icu_visit == "Y"].pat_key
```
Comment (beansrowning) [inline python/feature_tokenization.py:182]: @scotthlee The new targets file uses a string instead of a number for this column, so I had to swap this to populate it correctly

File: python/model.py [hunk 4]
```diff
@@ -321,15 +330,39 @@
 
         # Making the variables
         X = keras.preprocessing.sequence.pad_sequences(features,
-                                                       maxlen=225,
                                                        padding='post')
 
+        # DAN Model feeds in all features at once, so there's no need to limit to the
```
Comment (beansrowning) [inline python/model.py:335]: @scotthlee I mentioned it here, but the existing code padded to a hardcoded 225 (sequence length for the LSTM), but since the DAN is just flattening all the features into a single "bag", we can/should probably just pad to the largest bag size and then take that as the input shape.
